Explorer/src/components/sidebar.tsx

```diff
@@ -0,0 +1,120 @@
+"use client";
+
+import classNames from "classnames";
+import { Sidebar as FlowbiteSidebar } from "flowbite-react";
+import type { FC, PropsWithChildren } from "react";
+import { useSidebarContext } from "@/context/sidebar_context";
+import {
+  HiPresentationChartLine,
+  HiCog,
+  HiFolder,
+  HiServer,
+  HiTemplate,
+  HiChip,
+  HiTable,
+  HiHome,
+} from "react-icons/hi";
+
+const Sidebar: FC<PropsWithChildren<Record<string, unknown>>> = function ({
+  children,
+}) {
+  const { isOpenOnSmallScreens: isSidebarOpenOnSmallScreens } =
+    useSidebarContext();
+
+  return (
+    <div
+      className={classNames(
+        "fixed overflow-auto top-0 h-screen z-10 lg:sticky lg:!block",
+        {
+          hidden: !isSidebarOpenOnSmallScreens,
+        }
+      )}
+    >
+      <FlowbiteSidebar>{children}</FlowbiteSidebar>
+    </div>
+  );
+};
+
+export default Object.assign(Sidebar, { ...FlowbiteSidebar });
+
+import { usePathname } from "next/navigation";
+
+export function ActualSidebar(): JSX.Element {
+  const pathname = usePathname();
+  const splitPathname = pathname.split("/");
+  const current_page = splitPathname[1];
+
+  return (
+    <FlowbiteSidebar>
+      <FlowbiteSidebar.Items>
+        <FlowbiteSidebar.ItemGroup>
+          <FlowbiteSidebar.Item
+            href="dashboard"
+            icon={HiPresentationChartLine}
+            active={current_page.localeCompare("dashboard") ? false : true}
```

Comment:
nit: I don't like this use of the ternary.
Isn't this equivalent to this line?
```suggestion
            active={current_page === "dashboard"}
```